test/utils/mochaw.js

```diff
@@ -1,12 +1,26 @@
 import mocha from 'mocha';
-const {By} = require('selenium-webdriver');
+const {By, until} = require('selenium-webdriver');
 const expect = require('chai').expect
 
 const $driver = driver => selector =>
   driver.findElement(By.css(selector))
 
+const waitAndFindElement = driver => selector => {
+  const locator = By.css(selector)
+  return driver.findElement(async () => {
+    await driver.wait(until.elementLocated(locator))
+    return driver.findElement(locator)
+  })
+}
```

Comment:
`findAndWaitAndFind` 😸 

Was **this** crucial approach to solve the problem with tapping on the shutter button in the test?

Reply:
Yes. In the previous implementation the `takeSelfie()` function was firing as soon as the document was submitted. So in many cases, the driver was not able to locate the element because the upload was still in progress. With the wait, this function still fires straight after the previous step but the use of `wait/until` + `elementLocated` creates a loop until the element is located. This function will return the web element. Please not that it's better to call `isDisplayed()` over the returned element before trying to click on it. That's why I created a click wrapper function to make sure that every element is visible and is enabled before click is fired.